src/training/main.py

```diff
@@ -133,6 +133,22 @@ def main(args):
     )
     random_seed(args.seed, args.rank)
 
+    model_without_ddp = model
+    visual_ld = args.visual_ld if args.visual_ld else args.ld
+    text_ld = args.text_ld if args.text_ld else args.ld
+    
+    if visual_ld < 1.0:
+        visual_num_layers = model_without_ddp.visual.get_num_layers()
+        assigner_visual = LayerDecayValueAssigner(list(visual_ld ** (visual_num_layers + 1 - i) for i in range(visual_num_layers + 2)))
```

Comment:
consider using a more descriptive variable name here, explaining what is being assigned (e.g. `lr_assigner_visual`). Also is there any reason why the exponent logic is not inside the Assigner class? I.e. the constructor could take in the layer decay and number of layers, and compute the values accordingly

Reply:
+1 on that, plus getting this out of main